protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -83,22 +84,41 @@ func (rpccs *RPCConsumerServer) ServeRPCRequests(ctx context.Context, listenEndp
 	rpccs.finalizationConsensus = finalizationConsensus
 	rpccs.consumerAddress = consumerAddress
 	rpccs.consumerConsistency = consumerConsistency
+	initialRelays := true
+	rpccs.relaysMonitor = relaysMonitor
+	if RelaysHealthEnable {
+		rpccs.relaysMonitor.SetRelaySender(func() (bool, error) {
+			if initialRelays {
+				// Only start after everything is initialized - check consumer session manager
+				ok, err := rpccs.waitForPairing()
+				if !ok {
+					return false, err
+				}
+			}
 
-	chainListener, err := chainlib.NewChainListener(ctx, listenEndpoint, rpccs, rpcConsumerLogs, chainParser)
+			success, err := rpccs.sendCraftedRelays(MaxRelayRetries, initialRelays)
+			initialRelays = false
+			return success, err
+		})
+	}
+
+	chainListener, err := chainlib.NewChainListener(ctx, listenEndpoint, rpccs, rpccs, rpcConsumerLogs, chainParser)
 	if err != nil {
 		return err
 	}
+
 	go chainListener.Serve(ctx, cmdFlags)
-	// we trigger a latest block call to get some more information on our providers
-	go rpccs.sendInitialRelays(MaxRelayRetries)
+
+	// we trigger a latest block call to get some more information on our providers, using the relays monitor
+	if RelaysHealthEnable {
```

Comment:
we dont need twice the if "RelaysHealthEnable" case just do it all after the serve method?

Reply:
Fixed in 9faec8715f4d9950b16f25524f94484c0d48248c